### PR TITLE
Create Student Table (but no interface to view it yet)

### LINKED ISSUE #12: Create Student Table
In this Issue you will create a React component for displaying data from the /api/students/all endpoint.

You can populate that endpoint by using the /api/students/upload/egrades endpoint on swagger with data from /docs/examples/egrades.csv

The table should show the available fields from the  /api/students/all  endpoint.

# Acceptance Criteria

- [x] The React component described above exists
- [x] It has a storybook entry
- [x] It has test coverage

# Discussion

This will eventually be used on the Course Show Page #55

### PR DESCRIPTION
**Pull request by rtzhang1x**
In this PR, we created the React component for displaying student data from the /api/students/all endpoint and its corresponding tests and storybook. I have yet to create an interface to actaully display the data.

Closes #12.

### CODE REVIEW

File: frontend/src/tests/components/Student/StudentTable.test.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import StudentTable from "main/components/Student/StudentTable"
+import { currentUserFixtures } from "fixtures/currentUserFixtures";
+// import { fireEvent, render, waitFor, screen } from "@testing-library/react";
```
Comment (pconrad) [inline frontend/src/tests/components/Student/StudentTable.test.js]: Please remove commented out code:
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/tests/components/Student/StudentTable.test.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      const header = screen.getByTestId(`${testId}-cell-row-0-col-${field}`);
+      expect(header).toBeInTheDocument();
+    });
+
+    expect(screen.getByTestId(`${testId}-cell-row-0-col-id`)).toHaveTextContent("1");
+    expect(screen.getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+
+    // const deleteButton = screen.getByTestId(`${testId}-cell-row-0-col-Delete-button`);
+    // expect(deleteButton).toBeInTheDocument();
+    // expect(deleteButton).toHaveClass("btn-danger");
```
Comment (pconrad) [inline frontend/src/tests/components/Student/StudentTable.test.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by JUstISAGE — COMMENTED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Review by JUstISAGE — APPROVED** [review]
LGTM

**Pull request merged**